src/azure-cli/azure/cli/command_modules/storage/operations/blob.py

```diff
@@ -416,16 +417,62 @@ def generate_sas_blob_uri(client, container_name, blob_name, permission=None,
-        container_name, blob_name, permission=permission, expiry=expiry, start=start, id=id, ip=ip,
-        protocol=protocol, cache_control=cache_control, content_disposition=content_disposition,
-        content_encoding=content_encoding, content_language=content_language, content_type=content_type)
+                          content_type=None, full_uri=False, as_user=False):
+    if as_user:
+        user_delegation_key = client.get_user_delegation_key(
+            _get_datetime_from_string(start) if start else datetime.utcnow(), _get_datetime_from_string(expiry))
+        sas_token = client.generate_blob_shared_access_signature(
+            container_name, blob_name, permission=permission, expiry=expiry, start=start, id=id, ip=ip,
+            protocol=protocol, cache_control=cache_control, content_disposition=content_disposition,
+            content_encoding=content_encoding, content_language=content_language, content_type=content_type,
+            user_delegation_key=user_delegation_key)
+    else:
+        sas_token = client.generate_blob_shared_access_signature(
+            container_name, blob_name, permission=permission, expiry=expiry, start=start, id=id, ip=ip,
+            protocol=protocol, cache_control=cache_control, content_disposition=content_disposition,
+            content_encoding=content_encoding, content_language=content_language, content_type=content_type)
     if full_uri:
         return client.make_blob_url(container_name, blob_name, protocol=protocol, sas_token=sas_token)
     return sas_token
 
 
+def generate_container_shared_access_signature(client, container_name, permission=None,
+                                               expiry=None, start=None, id=None, ip=None,  # pylint: disable=redefined-builtin
+                                               protocol=None, cache_control=None, content_disposition=None,
+                                               content_encoding=None, content_language=None,
+                                               content_type=None, as_user=False):
+    '''
+    Generates a shared access signature for the container.
+    Use the returned signature with the sas_token parameter of any BlobService.
+
+    :param str cache_control:
+        Response header value for Cache-Control when resource is accessed
+        using this shared access signature.
+    :param str content_disposition:
+        Response header value for Content-Disposition when resource is accessed
+        using this shared access signature.
+    :param str content_encoding:
+        Response header value for Content-Encoding when resource is accessed
+        using this shared access signature.
+    :param str content_language:
+        Response header value for Content-Language when resource is accessed
+        using this shared access signature.
+    :param str content_type:
+        Response header value for Content-Type when resource is accessed
+        using this shared access signature.
+    '''
```

Comment:
Won't block the PR for this, but if this docstring exists to supply help text, that is actively discouraged. The help text should be in the `help.py`.

Reply:
Resolved.